10-methods.slide

```diff
@@ -0,0 +1,69 @@
+Methods
+22 Feb 2017
+Tags: edmonton-go, golang, workshop, workshop-one, functions, methods
+
+James Bell
+CEO, Bellstone Engineering
+james@bellstone.ca
+
+* Review: Functions
+
+.play 10-methods/printPerson.go /BEGIN OMIT/,/END OMIT/
+
+* Review: User-defined types
+
+.play 10-methods/userType.go /BEGIN OMIT/,/END OMIT/
+
+* Method on a type
+
+.play 10-methods/methodPrintPerson.go /BEGIN OMIT/,/END OMIT/
+
+* Methods can change data
+
+.play 10-methods/incrementor.go /BEGIN OMIT/,/END OMIT/
+
+* Methods can change data by using pointers
+
+- In order to change data, the method must act on a pointer.
+- The only necessary change to the `Incrementor` is to make the `Increment` function take a pointer.
+
+.play 10-methods/incrementorFixed.go /BEGIN OMIT/,/END OMIT/
+
+* Go converts between pointers and values automatically
+
+- On the last slide, an `Incrementor` value was used to call a pointer method.
+- Here, the variable `inc` is created as a pointer instead of as a value, but there's no need to change the rest of the code.
+
+.play 10-methods/incrementorPointer.go /BEGIN OMIT/,/END OMIT/
+
+* Scope
+
+- As with functions and variables, capitalized methods can be accessed from other packages.
+- The incrementor's `int` is private, but all of the methods and the type are public.
+
+.play 10-methods/incrementorFixed.go /func main/,/END OMIT/
+
+* Printing user types
+
+- The default printout of a type is usually not desired.
+
+.play 10-methods/printPersonDefault.go /BEGIN OMIT/,/END OMIT/
+
+* Printing user types: Stringer
```

Comment:
I'll remove those slides and see if they should be added to the interfaces section...

@mckayb24 , I have a couple of slides that might be useful for your presentation (about the Stringer interface). They refer to some types I've defined in my talk, but since your talk comes right after mine, I think people will remember well enough that it could work.
You could also use one of the exercises that I've now dropped: Write a stringer for the struct that was created for the exercises in talk 9. (This assumes that @torbiak adds an exercise to get the students to write a struct.)

Are the exercises supposed to be writing code like mine, or questions to think about? Most of the other writers have not included code exercises.